packages/data-stores/src/add-ons/use-add-on-checkout-link.ts

```diff
@@ -9,25 +8,26 @@ import { getSelectedSite } from 'calypso/state/ui/selectors';
  * @returns {Function} A function returnig a formatted checkout link for the given add-on and quantity
  */
 
-const useAddOnCheckoutLink = (): ( ( addOnSlug: string, quantity?: number ) => string ) => {
-	const selectedSite = useSelector( getSelectedSite );
+export const useAddOnCheckoutLink = (): ( (
+	siteDetails: SiteDetails | null,
```

Comment:
Thanks for the quick review! Will mull this over shortly 🤔

Reply:
Just getting back to this now 🙂

> I wonder if requiring the entire SiteDetails structure might turn out to be restrictive in the end for enabling reuse

Yep! I prefer your suggestion, since it's more explicit about what's actually necessary for the `useAddOnCheckoutLink` hook and makes debugging a little clearer ( as well as what you've already mentioned about reuse )

Making the changes now